### DISCUSSION TITLE
Does ruby-kafka supports on-demand consumers?

### DISCUSSION THREAD

**Discussion by tomerpe** [zendesk/ruby-kafka#899] [Q&A] [answered]
I want to implement a script that will route messages between specified Kafka topics.

I was wondering if `ruby-kafka` can provide this ability to basically create an on-demand consumer that will consume and produce all messages from a topic until he gets to some kind of an end (e.g. offset, eof, timestamp, etc.)? like `kafkacat` does...

I saw the `kafka.each_message` API method which can do the on-demand part without creating any consumer group but I cant see how I can stop it once it reaches to some kind of a breaking point.

**Comment by dasch** [comment, marked as answer]
That's not currently supported, but I'd be interested in proposals for an API and implementation of consume-until-end.
